.github/workflows/cron.yml

```diff
@@ -1,144 +1,197 @@
 name: Cron Build
 
-on: 
+on:
   schedule:
 #             ┌───────────── minute (0 - 59)
 #             │ ┌───────────── hour (0 - 23)
 #             │ │ ┌───────────── day of the month (1 - 31)
 #             │ │ │ ┌───────────── month (1 - 12 or JAN-DEC)
 #             │ │ │ │ ┌───────────── day of the week (0 - 6 or SUN-SAT)
-#             │ │ │ │ │                                   
+#             │ │ │ │ │
 #             │ │ │ │ │
 #             │ │ │ │ │
 #             * * * * *
     - cron:  '0 */6 * * *'
+  workflow_dispatch:
+    inputs:
+      idf_branch:
+        # This input can be an escaped JSON list of strings e.g. [\"release/v5.1\"] or "all" to build all branches.
+        # For IDF versions before v5.1, "all" is not supported. Use a specific branch instead.
+        description: 'IDF branch to build (check workflow file for instructions on how to use this input)'
+        required: true
+        default: 'all'
+      target:
+        # This input can be an escaped JSON list of strings e.g. [\"esp32\", \"esp32s2\"] or "all" to build all targets.
+        # For IDF versions before v5.1, "all" is not supported. Use a specific target list instead.
+        description: 'Target to build (check workflow file for instructions on how to use this input)'
+        required: true
+        default: 'all'
 
 defaults:
   run:
     shell: bash
 
 jobs:
-  run:
-    name: Build with IDF ${{ matrix.idf_branch }}
+  gen-matrix:
+    name: Generate matrix
     runs-on: ubuntu-latest
-    
+    outputs:
+      matrix: ${{ steps.gen-matrix.outputs.matrix }}
+      branches: ${{ steps.gen-matrix.outputs.branches }}
+    steps:
+      - name: Generate matrix
+        id: gen-matrix
+        run: |
```

Comment:
can this be a script?

Reply:
Yeah, didnt think about that. It will be even easier to maintain

Reply:
you marked as resolved, but this is still not a script, rather inline